File: lc1456.cs
public class Solution {
    public int MaxVowels(string s, int k) {
       int n = s.Length;
        int max = 0;
        int cur = 0;
        for(int i=0;i<n;i++)
        {
            if(i+1<=k)
            {
                if (s[i]=='a' || s[i]=='e' ||s[i]=='i' ||s[i]=='o' ||s[i]=='u')
                {
                    cur++;
                }
            }
            else
            {                
                max = Math.Max(max, cur);
                int j = i-k;
                //Console.WriteLine("i={0}, j={1}", i, j);
                if (s[j]=='a' || s[j]=='e' ||s[j]=='i' ||s[j]=='o' ||s[j]=='u')
                {
                    cur--;
                }
                if (s[i]=='a' || s[i]=='e' ||s[i]=='i' ||s[i]=='o' ||s[i]=='u')
                {
                    cur++;
                }
            }                
        }
        max = Math.Max(max, cur);
        return max;  
    }
}

File: java/lc1456.java
class Solution {
    public int maxVowels(String s, int k) {
        int n = s.length();
        int max = 0;
        int cur = 0;
        for(int i=0;i<n;i++)
        {
            if(i+1<=k)
            {//we are still building sliding window.
                if (s.charAt(i)=='a' || s.charAt(i)=='e' ||s.charAt(i)=='i' ||s.charAt(i)=='o' ||s.charAt(i)=='u')
                {
                    cur++;
                }
            }
            else
            {                
                max = Math.max(max, cur);
                int j = i-k; //oldest character index.
                if (s.charAt(j)=='a' || s.charAt(j)=='e' ||s.charAt(j)=='i' ||s.charAt(j)=='o' ||s.charAt(j)=='u')
                {
                    cur--;
                }
                if (s.charAt(i)=='a' || s.charAt(i)=='e' ||s.charAt(i)=='i' ||s.charAt(i)=='o' ||s.charAt(i)=='u')
                {
                    cur++;
                }
            }                
        }
        max = Math.max(max, cur);
        return max;  
    }
}
